Feat: Static Dataflow Analysis Framework and Visualization Integration

ArwinGit opened the pull request:
This Merge Request introduces Static Dataflow Analysis to While*. 

Features:
- CFG Generation: Generates an CFG with support for visual debugging via Mermaid.
- Visualisation Support (Wiz): (Mermaid with color options)
- Analysis Check (Added a check interface to validate analysis results)

Three concrete analysis have been implemented into the framework: Live Variables, Reaching Definitions and Reachability Analysis. (Live Variables and Reachability are displayable in wiz)

fhowar committed:
initial proposal

# Conflicts:
#	src/main/kotlin/tools/aqua/wvm/analysis/hoare/SMTSolver.kt
#	src/main/kotlin/tools/aqua/wvm/parser/Parser.kt

ArwinGit committed:
wiz implementations

haetze committed:
apply spotless

ArwinGit committed:
refactor: Improve readability and add assert scaffold

ArwinGit committed:
feat: Add new dataflow expressions, improve CFG gen, and refactor code

ArwinGit committed:
feat: adding Kill/Gen output + correct dataflow analysis

ArwinGit committed:
feat: adding reachability analysis, cfg string improvements, corrections

haetze committed:
applies spotless

haetze commented:
How can I start the dataflow analysis from the command line interface? I don't see the option in Main.kt. 

Also, please remember to run ```:/gradlew spotlessApply``` before committing. I did this time for you (please pull).

fhowar committed:
data flow analysis extension and checks

ArwinGit committed:
feat: Add dataflow analysis commands and applied spotless

Integrates liveness, reachability, reachingdefinitions and taint into the command-line interface.

ArwinGit committed:
feat: adding proof UI

ArwinGit committed:
feat: adding proof UI

haetze committed:
adds a hoare example with input

ArwinGit committed:
feat: adding objects for wiz analysis (Taint,RD)

fhowar committed:
rd like in slides

ArwinGit committed:
fix: taint check, rd improvements + correcting cfg generation

ArwinGit committed:
fix: small fixes/improvements (dataflow analysis)

fhowar committed:
fix HPC

fhowar committed:
quantification fix

fhowar committed:
HPC visualization

ArwinGit committed:
fix: adding missing expressions for dataflow analysis

haetze committed:
adds invariant to live/reachability analysis + completes varsInExpr for address expressions
